spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -1505,7 +1505,10 @@ void ParserBuilder::newValueForField(const production::Meta& meta, const Express
     if ( value.type().isA<type::Void>() ) {
         // Special-case: No value parsed, but still run hook.
         beforeHook();
-        builder()->addMemberCall(state().self, ID(fmt("__on_%s", field->id().local())), {}, field->meta());
+
+        if ( ! field->hooks().empty() )
```

Comment:
This won't work: we do always need to call hooks unconditionally because they could be implemented in a different translation unit. The bug must be somewhere else.